packages/block-editor/src/components/list-view/use-list-view-images.js

```diff
@@ -0,0 +1,79 @@
+/**
+ * WordPress dependencies
+ */
+import { useMemo } from '@wordpress/element';
+import { useSelect } from '@wordpress/data';
+
+/**
+ * Internal dependencies
+ */
+import { store as blockEditorStore } from '../../store';
+
+// Maximum number of images to display in a list view row.
+const MAX_IMAGES = 5;
+
+function getImageUrl( block ) {
+	if ( block.name !== 'core/image' ) {
+		return;
+	}
+
+	if ( block.attributes?.url ) {
+		return { url: block.attributes.url, alt: block.attributes.alt };
+	}
+}
+
+function getImagesFromGallery( block ) {
```

Comment:
Would it make sense to expand this to other container blocks that may have images, such as Group or Column? (Not necessarily as part of this PR)

Reply:
Yes — I was thinking we'd start with Gallery + Image for now, and look at other blocks such as Cover, etc in a follow-up where we can look at how an API for it might work, as in [the other discussion](https://github.com/WordPress/gutenberg/pull/53381#discussion_r1285973317) above.

I like the idea of Group, Column, or other container blocks also showing what's in them, that's a cool idea! And if the API discussion winds up being a bit complex, then it'll be a simple follow-up to add hard-coded rules to allow Group, Column, etc in, too.

Reply:
We can try it, but it may be more distracting than necessary to have images also visible in parent group/columns blocks, and not just image/media related blocks.

Reply:
Wrote a follow-up issue to add support for other image-attribute-based blocks: https://github.com/WordPress/gutenberg/issues/53684

Reply:
Nice, that sounds good to me 👍